execution_engine/src/storage/global_state/scratch.rs

```
@@ -108,7 +114,12 @@ impl StateReader<Key, StoredValue> for ScratchGlobalStateView {
         correlation_id: CorrelationId,
         key: &Key,
     ) -> Result<Option<StoredValue>, Self::Error> {
-        if let Some(value) = self.cache.read().unwrap().get(key) {
+        if let Some(value) = self
+            .cache
+            .read()
+            .expect("Cache lock should not be poisoned")
```

We can throw an error here...